scenarioo-client/app/manage/label-colors/color-helpers.ts

```diff
@@ -0,0 +1,54 @@
+export const isHexColorValid = (color: string): boolean => {
+    // Color hexadecimal should have the format #000 or #0000000
+    const hexadecimalColorRegexPattern = /^#([0-9A-F]{3}){1,2}$/i;
+    return hexadecimalColorRegexPattern.test(color);
+};
+
+export const getRandomHexColor = (): string => {
+    const hexChars: string = '0123456789ABCDEF';
+    const numberOfChars: number = 6;
+    let randomHexColor: string = '#';
+    for (let i = 0; i < numberOfChars; i++) {
+        randomHexColor += hexChars[Math.floor(Math.random() * 16)];
+    }
+    return randomHexColor;
+};
+
+export const getContrastingColor = (hexColor: string): string => {
+    const lightColor = '#FFFFFF';
+    const darkColor = '#000000';
+    const [r, g, b] = [...hexToRgb(hexColor)];
+    // Determine the contrast of the color by calculating the brightness using YIQ values
+    const brightness = (299 * r + 587 * g + 114 * b) / 1000;
+    if (brightness > 127.5) {
+        return darkColor;
+    }
+    return lightColor;
+};
+
+const hexToRgb = (hexColor: string): number[] => {
+    hexColor = extendColorHex(hexColor);
+
+    hexColor = hexColor.replace('#', '');
+
+    const r: number = parseInt(hexColor.slice(0, 2), 16);
+    const g: number = parseInt(hexColor.slice(2, 4), 16);
+    const b: number = parseInt(hexColor.slice(4, 6), 16);
+    return [r, g, b];
+};
+
+const extendColorHex = (hexColor: string): string => {
+    const hexGoalLength: number = 6;
+    const hexColorValue: string = hexColor.replace('#', '');
+    if (hexColorValue.length === hexGoalLength) {
+      return hexColor;
+    }
+
+    // Convert e.g. #ABC to #AABBCC
+    return '#' + hexColorValue
+        .split('')
+        .reduce((extendedHexColor: string[], currentHexValue: string) => {
+            return extendedHexColor.concat([currentHexValue, currentHexValue]);
+        }, [])
+        .join('');
+};
```

Comment:
Würde daraus eine Klasse mit statischen Methoden machen und sie Color nennen - Color.isValid(...), Color.getRandom(), und Color.getContrasting(..) sieht hübsch aus und lässt sich einfacher importieren als wenn ich die genauen Methodennamen erst raussuchen und einzeln importieren muss.

Reply:
diese Utility klasse würde ich in ein Utility package legen, falls wir noch mehr mit Farben machen

Reply:
This makes sense and does simplify the import. Thank you for pointing it out.
I've created a Color class with static functions and placed it in the shared/utils folder.